xcode/Subconscious/Shared/Services/Noosphere/NoosphereService.swift

```diff
@@ -407,8 +410,20 @@ actor NoosphereService:
         try await self.sphere().traverse(petname: petname)
     }
     
-    func sphere(did: Did) async throws -> Sphere {
-        try Sphere(noosphere: self.noosphere(), identity: did.did)
+    /// Intelligently open a sphere by traversing or, if this is our address, returning the default sphere.
+    func sphere(address: Slashlink) async throws -> Sphere {
+        let identity = try await self.identity()
+        
+        switch (address.peer) {
+        case .none:
+            return try self.sphere()
+        case .some(.did(let did)) where did == identity:
+            return try self.sphere()
```

Comment:
Annoyingly I can't merge these two cases because `did` is only bound in one of them?

Unless I missed some syntax for where clauses + multiple cases.

Reply:
Nit: you can omit `.some(...)` and just say `.did(let did) where did == identity`